packages/cli/src/commands/verify/spec.ts

```diff
@@ -0,0 +1,129 @@
+import { ParamDetails, Arg, Option } from '../../register-command';
+
+import { TxParams } from '@openzeppelin/upgrades';
+import NetworkFile from '../../models/files/NetworkFile';
+import { DEFAULT_TX_TIMEOUT } from '../../models/network/defaults';
+
+import ProjectFile from '../../models/files/ProjectFile';
+
+export const name = 'verify';
+export const description = "verify a contract's source with Etherscan or Etherchain";
+
+export interface Args {
+  contract: string;
+}
+
+type OptimizerOptions = { optimizer: false } | { optimizer: true; optimizerRuns: number };
+
+type RemoteOptions = { remote: 'etherscan'; apiKey: string } | { remote: 'etherchain' };
+const endpoints = ['etherscan', 'etherchain'];
+
+interface OtherOptions {
+  network: string;
+
+  // The following are not available as CLI flags, and they are only used in tests.
+  networkFile?: NetworkFile;
+  txParams?: TxParams;
+}
+
+const compilerConfig = new ProjectFile().compilerOptions;
+const compilerDefaults = {
+  optimizer: (compilerConfig.optimizer && compilerConfig.optimizer.enabled) || false,
+  optimizerRuns: (compilerConfig.optimizer && compilerConfig.optimizer.runs) || 200,
+};
+
+export type Options = OptimizerOptions & RemoteOptions & OtherOptions;
+
+export const args: Arg[] = [
+  {
+    name: 'contract',
+    async details(): Promise<ParamDetails> {
+      const choices = await import('../../prompts/choices');
+
+      const contracts = choices.contracts('all');
+
+      return {
+        prompt: 'Pick a contract to verify',
+        choices: contracts,
+      };
+    },
+  },
+];
+
+export const options: Option[] = [
```

Comment:
This is missing the `--no-interactive` option. Can you take the one from `deploy` and put it in a place where it can be reused here?

Reply:
It's not missing it, it's just at the very end (which I find leads to a better `--help` message). I'll add a file with command utils.

Reply:
I added a simple `utils` file with this declaration: I suspect we'll want to reuse not only options but also common validations, such as checking if a value is an address, a number, etc.

Note we could also almost remove the duplication of the `network` option, but they are different due to the `after` in deploy. I thought about making this parametrized, but didn't want to complicate things unnecessarily. Up to you!